### PR TITLE
NetworkPolicy Issue: KeyError: 'default:default-allow-egress_egress_0'

### LINKED ISSUE #397: NetworkPolicy Issue: KeyError: 'default:default-allow-egress_egress_0'
Repro:
kubectl run pod0 --image=fwnetworking/testpod
kubectl run pod1 --image=fwnetworking/testpod

kubectl apply -f  /home/ubuntu/mizar/mizar/obj/tests/test_networkpolicy.yaml
kubectl apply -f /home/ubuntu/go/src/k8s.io/mizar/mizar/obj/tests/test_networkpolicy_default_allow_ingress.yaml
kubectl apply -f /home/ubuntu/go/src/k8s.io/mizar/mizar/obj/tests/test_networkpolicy_default_allow_egress.yaml

kubectl delete networkpolicies default-allow-ingress

Got:
[2021-01-19 09:41:42,979] luigi-interface      [ERROR   ] [pid 7] Worker Worker(salt=957853790, workers=1, host=kind-worker3, username=root, pid=7) failed    k8sNetworkPolicyDelete(param=<mizar.common.wf_param.HandlerParam object at 0x7f4241a78a00>)
Traceback (most recent call last):
  File "/usr/local/lib/python3.8/site-packages/luigi/worker.py", line 199, in run
    new_deps = self._run_get_new_deps()
  File "/usr/local/lib/python3.8/site-packages/luigi/worker.py", line 141, in _run_get_new_deps
    task_gen = self.task.run()
  File "/usr/local/lib/python3.8/site-packages/mizar/dp/mizar/workflows/builtins/networkpolicies/delete.py", line 42, in run
    networkpolicy_util.handle_networkpolicy_update_delete(endpoint_names)
  File "/usr/local/lib/python3.8/site-packages/mizar/networkpolicy/networkpolicy_util.py", line 137, in handle_networkpolicy_update_delete
    self.handle_endpoint_for_networkpolicy(ep)
  File "/usr/local/lib/python3.8/site-packages/mizar/networkpolicy/networkpolicy_util.py", line 140, in handle_endpoint_for_networkpolicy
    data_for_networkpolicy_ingress = self.generate_data_for_networkpolicy_ingress(ep)
  File "/usr/local/lib/python3.8/site-packages/mizar/networkpolicy/networkpolicy_util.py", line 178, in generate_data_for_networkpolicy_ingress
    self.build_access_rules(data, ep)
  File "/usr/local/lib/python3.8/site-packages/mizar/networkpolicy/networkpolicy_util.py", line 312, in build_access_rules
    self.build_cidr_table(access_rules, ep, "no_except")
  File "/usr/local/lib/python3.8/site-packages/mizar/networkpolicy/networkpolicy_util.py", line 362, in build_cidr_table
    "bit_value": self.calculate_policy_bit_value(access_rules, indexed_policy_names),
  File "/usr/local/lib/python3.8/site-packages/mizar/networkpolicy/networkpolicy_util.py", line 370, in calculate_policy_bit_value
    policy_bit_value += access_rules["indexed_policy_map"][indexed_policy_name]
KeyError: 'default:default-allow-egress_egress_0'

### PR DESCRIPTION
**Pull request by Hong-Chang**

### CODE REVIEW

File: mizar/networkpolicy/networkpolicy_util.py [hunk 1]
```diff
@@ -327,7 +327,8 @@ def build_cidr_and_policies_map(self, access_rules, cidr_type):
             for found_cidr_tuple in found_cidr_map:
                 if indexed_policy_names != found_cidr_tuple[1]:
                     for foundPolicyName in found_cidr_tuple[1]:
-                        indexed_policy_names.add(foundPolicyName)
+                        if cidr in access_rules[cidr_and_policies_map_name] and foundPolicyName in access_rules[cidr_and_policies_map_name][cidr]:
```
Comment (h-w-chen) [inline mizar/networkpolicy/networkpolicy_util.py:332]: not sure why such (added) condition is needed. if some explanation is put here, that would help readers to understand.

File: mizar/obj/endpoint.py [hunk 2]
```diff
@@ -462,6 +462,8 @@ def delete_network_policy_egress(self, cidr_type, cidr_table):
     def delete_network_policy_protocol_port_ingress(self, port_table):
         cidr_networkpolicy_list = []
         for item in port_table:
+            if item["protocol"] == "any" and item["port"] == "0":
```
Comment (h-w-chen) [inline mizar/obj/endpoint.py]: does this block mean the open-to-all policy never been deleted?

**Review by h-w-chen — CHANGES_REQUESTED** [review]
This PR is nicely focused. If the reason of the change were made clear to reader, that would help reader to comprehend; one of the ways is description of PR to briefly state its purpose.

File: mizar/obj/endpoint.py (hunk 2 above)
Comment (Hong-Chang) [reply mizar/obj/endpoint.py]: You are right. I withdrawal the change here..

File: mizar/networkpolicy/networkpolicy_util.py (hunk 1 above)
Comment (Hong-Chang) [reply mizar/networkpolicy/networkpolicy_util.py:332]: Added comments in code to explain why adding the line.

**Review by h-w-chen — APPROVED** [review]
lgtm

**Pull request merged**